prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -454,6 +453,20 @@ class ThinkThoroughlyWithPredictionProphetResearch(ThinkThoroughlyBase):
     model = "gpt-4-turbo-2024-04-09"
     model_for_generate_prediction_for_one_outcome = "gpt-4o-2024-08-06"
 
+    @staticmethod
+    def build_pydantic_ai_agent(
+        model: KnownModelName, temperature: float = 0.7
+    ) -> PydanticAIAgent:
+        api_keys = APIKeys()
+        provider = OpenAIProvider(api_key=api_keys.openai_api_key.get_secret_value())
+        return PydanticAIAgent(
+            OpenAIModel(
+                model,
+                provider=provider,
```

Comment:
Use `get_openai_provider` from latest PMA please, otherwise, we will be getting false-positive error logs.